src/mlagility/api/run_ort_model.py

```diff
@@ -0,0 +1,87 @@
+import argparse
+import re
+import math
+from timeit import default_timer as timer
+import numpy as np
+import onnxruntime as ort
+
+def run_ort_profile(onnx_file, num_iterations=100):
+    # Run the provided onnx model using onnxruntime and measure average latency
+
+    per_iteration_latency = []
+    sess_options = ort.SessionOptions()
+    sess_options.graph_optimization_level = ort.GraphOptimizationLevel.ORT_ENABLE_ALL
+    onnx_session = ort.InferenceSession(onnx_file, sess_options)
+    sess_input = onnx_session.get_inputs()
+    input_feed = _dummy_inputs(sess_input)
+    output_name = onnx_session.get_outputs()[0].name
+
+    for _ in range(num_iterations):
+        start = timer()
+        onnx_session.run([output_name], input_feed)
+        end = timer()
+        iteration_latency = end - start
+        per_iteration_latency.append(iteration_latency)
+
+    print (per_iteration_latency)
+
+
+def _dummy_inputs(sess_input) -> dict:
+    # Generate dummy inputs of the expected shape and type for the input model
+    input_stats = []
+    for _idx, input_ in enumerate(range(len(sess_input))):
+        input_name = sess_input[input_].name
+        input_shape = sess_input[input_].shape
+
+        # TODO: Use onnx update_inputs_outputs_dims to automatically freeze models
+        for dim in input_shape:
+            if isinstance(dim, str) is True or math.isnan(dim) is True:
+                raise AssertionError(
+                    "Error: Model has dynamic inputs. Freeze the graph and try again"
+                )
+
+        input_type = sess_input[input_].type
+        input_stats.append([input_name, input_shape, input_type])
+
+    input_feed = {}
+    for stat in input_stats:
+        dtype_str = re.search(r"\((.*)\)", stat[2])
+        assert dtype_str is not None
+        datatype = dtype_ort2str(dtype_str.group(1))
+        input_feed[stat[0]] = np.random.rand(*stat[1]).astype(datatype)
+    return input_feed
```

Comment:
Shouldn't we have a central place to generate the inputs for `ort`, `trt`, `groq`, etc? How do we avoid inconsistencies?

Reply:
The required input format differs for `trt` and `ort`. 
`trtexec` generates its own dummy inputs.
While `ort` requires input in a specific dict format. 
Since we only use these inputs for performance benchmarking on `trt` and `ort`, I don't think they need to be centralized.